documentation/specs/multithreading/analyzer-spec.md

```diff
@@ -0,0 +1,219 @@
+# IMultiThreadableTask Analyzer - Design Proposal
+
+**Target MSBuild Version**: 18.1+  
+**Authors**: @JanProvaznik  
+**Last Updated**: October 1, 2025
+
+---
+
+## Executive Summary
+
+### Problem
+
+MSBuild's multithreaded execution allows tasks implementing `IMultiThreadableTask` to run concurrently. Many .NET APIs depend on process-global state (working directory, environment variables), creating race conditions:
```

Comment:
I might broaden this a bit to something like "some things are bad practice to do in tasks, so let's have an analyzer that can help users write tasks. The first things we'll focus on will be the threading issues"